kebechet/managers/label_bot/label_bot.py

```diff
@@ -0,0 +1,90 @@
+
+import logging
+from kebechet.managers.manager import ManagerBase
+import typing
+import os
+import requests
+import json
+
+_LOGGER = logging.getLogger(__name__)
+
+# Github and Gitlab events on which the manager acts upon.
+_EVENTS_SUPPORTED = ["issues", "issue"]
+
+_GITHUB_LABEL_BOT_API = os.getenv("LABELBOT_URL")
+_MINIMUM_CONFIDENCE = 0.5
+_ISSUES_TO_IGNORE = [
+    "Kebechet",
+    "new patch release",
+    "new minor release",
+    "new major release",
+    "new calendar release",
+    "new pre-release",
+    "new build release",
+]
+
+
+class ThothLabelBotManager(ManagerBase):
+    """Labels issue using Thoth Github Issue classifier."""
+
+    def assign_label(self, response_dict: dict) -> typing.Tuple[typing.Any, float]:
+        """Return the label with the highest confidence in the response."""
+        label_confidence = []
+        for key in response_dict.keys():
+            if key not in ["title", "body"]:
+                label_confidence.append((key, float(response_dict[key])))
+        label_confidence.sort(key=lambda x: x[1], reverse=True)
+        return label_confidence[0]
+
+    def run(self) -> typing.Optional[dict]:  # type: ignore
+        """Check for info issue and close it with a report."""
+        if self.parsed_payload:
+            if self.parsed_payload.get("event") not in _EVENTS_SUPPORTED:
+                _LOGGER.info(
+                    "Label manager doesn't act on %r events.",
+                    self.parsed_payload.get("event"),
+                )
+                return None
+            # Note this manager is currently only supported on Github.
+            if (
+                self.parsed_payload.get("service_type") != "github"
+                or _GITHUB_LABEL_BOT_API is None
+            ):
+                _LOGGER.info("Label manager doesn't act on non github services.")
+                return None
+
+        payload = self.parsed_payload.get("raw_payload").get("payload")  # type: ignore
+        if payload.get("action") == "opened":
+            issue_title = payload.get("issue").get("title")
+            issue = self.sm.get_issue(issue_title)
+            if not issue:
+                _LOGGER.info("Issue not found, exiting")
+                return None
+
+            # TODO: Read from thoth.yaml so that user could add more issues to ignore.
+            for issue_to_ignore in _ISSUES_TO_IGNORE:
+                if issue_title.startswith(issue_to_ignore):
+                    _LOGGER.info("Ignored as it is a BOT related issue.")
+                    return None
+
+            _LOGGER.info(f"Found issue {issue_title}, predicting label.")
+            url = _GITHUB_LABEL_BOT_API + "/predict"  # type: ignore
+            headers = {"Content-type": "application/json", "Accept": "text/plain"}
+            data = {"title": str(issue.title), "body": str(issue.description)}
+            response = requests.request(
+                "POST", url, headers=headers, data=json.dumps(data)
```

Comment:
You can let requests to construct headers for JSON, you just need to pass:

```suggestion
                "POST", url, data=data
```

In that case, headers and JSON serialization is left on requests.

Reply:
I remember now the requests kept throwing this particular error, - 
```
Traceback (most recent call last):
  File "/home/sai/Downloads/kebechet/kebechet/config.py", line 166, in run_url
    instance.run(**manager_configuration)
  File "/home/sai/Downloads/kebechet/kebechet/managers/label_bot/label_bot.py", line 101, in run
    response_dict = response.json()
  File "/home/sai/.local/share/virtualenvs/kebechet-D6H2uQdh/lib/python3.8/site-packages/requests/models.py", line 898, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0
```

Reply:
That's why I stuck to doing serialization manually

Reply:
It throws an error on the [server side](https://github.com/thoth-station/Github-Issues-Classifier/blob/fbc80b28e3e480a00af733cb35a95ad47c5ca1c9/server.py#L47) - 
```
Needs to call the api of labelbot and assign a label to the issue.  Improvement add a label bot manager to kebechet
ERROR:tornado.application:Uncaught exception POST /predict (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:8888', method='POST', uri='/predict', version='HTTP/1.1', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/sai/.local/share/virtualenvs/Github-Issues-Classifier-caNbId4v/lib/python3.8/site-packages/tornado/web.py", line 1704, in _execute
    result = await result
  File "server.py", line 47, in post
    req = json_decode(self.request.body)
  File "/home/sai/.local/share/virtualenvs/Github-Issues-Classifier-caNbId4v/lib/python3.8/site-packages/tornado/escape.py", line 83, in json_decode
    return json.loads(to_basestring(value))
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
That's why I stuck to manually serializing it.